[neyapraveen] iP

neyapraveen opened the pull request:
### BotPro
“Your mind is for having ideas, not holding them.” – David Allen [([source])](https://dansilvestre.com/productivity-quotes)

BotPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- _FAST_ SUPER FAST to use
- All you need to do is,

1. download it from [[here].](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE!**

Features:
- [ ]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

neyapraveen committed:
Rename, Greet, Exit

neyapraveen committed:
Echo

neyapraveen committed:
Adds input to array list

neyapraveen committed:
Add, List

neyapraveen committed:
Created Task class

neyapraveen committed:
Mark as Done

neyapraveen committed:
Create Todo class

neyapraveen committed:
Create Deadline class

neyapraveen committed:
Create Event class

neyapraveen committed:
Create ToDos, Events, Deadlines

neyapraveen committed:
Create Test

neyapraveen committed:
Create DukeException class

neyapraveen committed:
Handle errors

neyapraveen committed:
Handle errors

neyapraveen committed:
Handle errors final

neyapraveen committed:
Delete Tasks

neyapraveen committed:
Create enum for TaskStatus

neyapraveen committed:
Save event to text file

wasjoe1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+                            }
+                            continue;
+                        } catch (NumberFormatException e) {
+                            System.err.println(Duke.horizontalLine + "You did not enter a valid integer :(\n" + Duke.horizontalLine);
+                            continue;
+                        }
+                    }
+                    if (instruction.equals("delete")) {
+                        try {
+                            int no = Integer.parseInt(elements[1]);
+                            Task taskInstance = new Task("");
+                            taskInstance.delete(no);
+                            continue;
+                        } catch (NumberFormatException e) {
+                            System.err.println(Duke.horizontalLine + "You did not enter a valid integer :(\n" + Duke.horizontalLine);
```
Was wondering if this line is longer than 120 words (as per the limit in our java coding standard)?

wasjoe1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,99 @@
+                    else if (instruction.equals("deadline")) {
+                        String[] taskAndDeadline = actualTask.split(("/by"));
+                        if (taskAndDeadline.length == 1 || taskAndDeadline.length == 0) {
+                            throw new DukeException(Duke.horizontalLine + "OOPS!!! Invalid format for deadline :-(\n" + Duke.horizontalLine);
```
Was wondering if this line is longer than 120 words (I notice this same issue in several other areas)?

wasjoe1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,70 @@
+
+public class Task {
+    private String task;
+    private TaskStatus status;
```
was wondering if using `isDone` or some other naming would make this clearer on whether the status of the task is done or not?

wasjoe1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,70 @@
+        counter = counter - 1;
+        arr.remove(i - 1);
+        System.out.println(Duke.horizontalLine + "Noted. I've removed this task:\n" + deleteTask.toString()
+        + "\n" + String.format("Now you have %d tasks in the list\n", counter) + Duke.horizontalLine );
```
I like that your + operator is in the next line.

wasjoe1 reviewed (commented):
I like how your code looks generally okay!
Only some nits from my end - the line formatting that passes 120 words. Otherwise, LGTM!

neyapraveen committed:
Save all tasks to text file

neyapraveen committed:
Merge branch 'branch-Level-7'

neyapraveen committed:
Handle Date and Time

neyapraveen committed:
Handle errors in date formats

neyapraveen committed:
Correct event constructor

neyapraveen committed:
Merge branch 'branch-Level-8'

tiif commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+
+    public void print() {
```
Would it be better to add more context to the method name like printAddEvent?

tiif commented on src/main/java/Task.java:
LGTM. It is following the coding standard.

tiif commented on src/main/java/Deadline.java:
LGTM. It is following the coding standard.

tiif commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,288 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    public static String horizontalLine = "_".repeat(60) + "\n";
```
I like how this is implemented.

neyapraveen committed:
Create Ui class

neyapraveen committed:
Create Storage class

neyapraveen committed:
Create TaskList class

neyapraveen committed:
Create Parser class

neyapraveen committed:
Implement more OOP

neyapraveen committed:
Merge branch 'A-MoreOOP'

* A-MoreOOP:
  Implement more OOP
  Create Parser class
  Create TaskList class
  Create Storage class
  Create Ui class

neyapraveen committed:
Divide classes into packages

neyapraveen committed:
Merge branch 'A-Packages'

* A-Packages:
  Divide classes into packages

neyapraveen committed:
Merge remote-tracking branch 'origin/add-gradle-support'

* origin/add-gradle-support:
  Bump gradle and lib version
  Add Gradle support

neyapraveen committed:
Automate project build using Gradle

neyapraveen committed:
Add JUnit test for Todo class

neyapraveen committed:
Add JUnit test for Deadline class

neyapraveen committed:
Merge remote-tracking branch 'origin/A-JUnit'

* origin/A-JUnit:
  Add JUnit test for Deadline class
  Add JUnit test for Todo class

neyapraveen committed:
Fix coding standard errors

neyapraveen committed:
Add JavaDoc comments

neyapraveen committed:
Tweak code to comply with java coding standard

neyapraveen committed:
Implement find function

neyapraveen committed:
Merge remote-tracking branch 'origin/A-JavaDoc'

* origin/A-JavaDoc:
  Add JavaDoc comments

neyapraveen committed:
Merge remote-tracking branch 'origin/A-CodingStandard'

* origin/A-CodingStandard:
  Tweak code to comply with java coding standard

# Conflicts:
#	src/main/java/duke/Event.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/Todo.java

neyapraveen committed:
Merge remote-tracking branch 'origin/branch-Level-9'

* origin/branch-Level-9:
  Implement find function

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java

neyapraveen committed:
Fix merge conflicts

neyapraveen committed:
Fix bugs in printing event and todo

neyapraveen committed:
Add basic GUI

neyapraveen committed:
Add functioning GUI

neyapraveen committed:
Fix styling in displaySearchResults method

neyapraveen committed:
Add auto greeting

neyapraveen committed:
Tidy up classes

neyapraveen committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Tidy up classes
  Add auto greeting
  Fix styling in displaySearchResults method

neyapraveen committed:
Exit application automatically with bye instruction

neyapraveen committed:
Fix bugs

neyapraveen committed:
Add assertions

neyapraveen committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  Add assertions

neyapraveen committed:
Shorten loadTaskFromFile method

loadTaskFromFile method needs to be shortened
and simplified to ensure greater readability and
standardised levels of abstraction. This makes
it challenging for people to understand the code.

Let’s create a createTask method that helps
to abstract out details specific to the creation
of tasks and handle them in another method.

This improves readability.

neyapraveen committed:
Shorten loadTaskFromFile method

loadTaskFromFile method needs to be shortened
And simplified to ensure greater readability and
standardised levels of abstraction. This makes
It challenging for people to understand the code.

Let’s create a updateTaskList method that helps
to abstract out details specific to the updating
of the task list when we load the data.

This improves readability and simplifies the code.

neyapraveen committed:
Shorten parse method in Parser class

neyapraveen committed:
Add javadoc to new methods

neyapraveen committed:
Merge pull request #2 from neyapraveen/branch-A-CodeQuality

Branch A-code-quality

neyapraveen committed:
Merge branches 'master' and 'master' of https://github.com/neyapraveen/ip

* 'master' of https://github.com/neyapraveen/ip:
  Add javadoc to new methods
  Shorten parse method in Parser class
  Shorten loadTaskFromFile method
  Shorten loadTaskFromFile method

* 'master' of https://github.com/neyapraveen/ip:
  Add javadoc to new methods
  Shorten parse method in Parser class
  Shorten loadTaskFromFile method
  Shorten loadTaskFromFile method

neyapraveen committed:
Edit javadocs

neyapraveen committed:
Make search function case insensitive

neyapraveen committed:
Merge pull request #3 from neyapraveen/branch-C-BetterSearch

Make search function case insensitive

neyapraveen committed:
Merge branch 'master' of https://github.com/neyapraveen/ip

* 'master' of https://github.com/neyapraveen/ip:
  Make search function case insensitive

neyapraveen committed:
Add colour to GUI and dialog box

neyapraveen committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Add colour to GUI and dialog box

neyapraveen committed:
Add help function

neyapraveen committed:
Merge branch 'branch-C-Help'

* branch-C-Help:
  Add help function

neyapraveen committed:
Add personality

neyapraveen committed:
Merge branch 'branch-A-Personality'

* branch-A-Personality:
  Add personality

neyapraveen committed:
Add a User Guide

neyapraveen committed:
Update link to release

neyapraveen committed:
Edit help message

neyapraveen committed:
Correct build config

neyapraveen committed:
Edit Quick Start

neyapraveen committed:
Update README.md

neyapraveen committed:
Update README.md
